syntax = "proto2";

package silver_bullet;

enum Mode {
    // Don't make any changes to this package
    skip = 0;

    // Push changes directly to the packaging repository
    push = 1;

    // Propose the changes in a pull request
    propose = 2;

    // Attempt to push the changes, but if that fails, create a pull request
    // instead.
    attempt_push = 3;
}

enum ChangelogMode {
    auto = 0;

    update_changelog = 1;

    leave_changlog = 2;
}

message Match {
    // Maintainer e-mail
    repeated string maintainer = 1;

    // Uploader e-mail
    repeated string uploader = 2;

    // Source package name
    repeated string source_package = 3;
}

message Policy {
    repeated Match match = 1;

    // Mode
    optional Mode mode = 2;

    // Changelog update mode
    optional ChangelogMode changelog = 3;

    // Fixers to apply
    repeated string lintian_fixers = 4;
}

message PolicyConfig {
    repeated Policy policy = 1;
}
